### PR TITLE
Omnipool stableswap arbitrage

### PR DESCRIPTION
**Pull request by jepidoptera**

### CODE REVIEW

File: hydradx/tests/test_arbitrage_agent_refactor.py [hunk 1]
```diff
@@ -110,7 +110,7 @@ def test_calculate_arb_amount_bid(
         sell_ex_max_sell=init_agent.holdings[tkn],
         buy_ex_max_sell=init_agent.holdings[numeraire],
         precision=p,
-        max_iters=100
+        max_iters=110
```
Comment (poliwop) [inline hydradx/tests/test_arbitrage_agent_refactor.py:113]: This seems really high, was this change really needed to get sufficient accuracy?

File: hydradx/notebooks/Misc/stableswap arb.ipynb [hunk 2]
```diff
@@ -0,0 +1,304 @@
...
+    "\n",
+    "import json"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "id": "fc7e81433023f150",
+   "metadata": {},
+   "source": [
+    "Set `get_live_markets = True` to get live markets from Kraken. Set to `False` to use order books from file."
```
Comment (poliwop) [inline hydradx/notebooks/Misc/stableswap arb.ipynb:36]: notebook appears to pull from Binance too?

**Review by poliwop — APPROVED** [review]
See comments for small changes you may want to make, but they are not required. This looks good.

File: hydradx/model/amm/agents.py [hunk 3]
```diff
@@ -47,8 +47,14 @@ def __repr__(self):
...
-        copy_self = copy.deepcopy(self)
-        return copy_self
+        # copy_self = Agent(
+        #     holdings={k: v for k, v in self.holdings.items()},
+        #     share_prices={k: v for k, v in self.share_prices.items()},
+        #     delta_r={k: v for k, v in self.delta_r.items()},
+        #     trade_strategy=self.trade_strategy,
+        #     unique_id=self.unique_id
+        # )
+        return copy.deepcopy(self)
```
[the remark this answers is not in the document]
Comment (jepidoptera) [reply hydradx/model/amm/agents.py]: Undoubtedly it does, but you were concerned about maintainability, and I thought that was valid.

File: hydradx/tests/test_arbitrage_agent_refactor.py (hunk 1 above)
Comment (jepidoptera) [reply hydradx/tests/test_arbitrage_agent_refactor.py:113]: This came about because I was getting one of those "surprise" test fails (where a test that had been passing suddenly doesn't). Long story short, I forgot to change it back after I figured out what the problem actually was. I'll change it back.
